### PR TITLE
Python Client constructor update

### PR DESCRIPTION
**Pull request by billschereriii**
Clean up Python Client constructor to detect invalid kwargs and remove deprecation warning for construction method that will be used primarily for driver scripts. Rename internal construction methods to reflect new expected usage patterns. Update test cases to remove invalid construction requests. Update docs.

### CODE REVIEW

File: src/python/module/smartredis/client.py [hunk 1]
```diff
@@ -121,7 +121,7 @@ def __deprecated_construction(self, cluster, address=None, logger_name="Default"
         except (PybindRedisReplyError, RuntimeError) as e:
             raise RedisConnectionError(str(e)) from None
 
-    def __new_construction(self, config_options=None, logger_name="Default"):  # pylint: disable=no-self-use
+    def __standard_construction(self, config_options=None, logger_name="Default"):  # pylint: disable=no-self-use
```
Comment (MattToast) [inline src/python/module/smartredis/client.py:124]: Not directly related this PR, but thought I should bring it up while we are here:

You should be able to remove the `# pylint: disable=...` comment by simply decorating this method with `@staticmethod` and removing the `self` parameter

File: src/python/module/smartredis/client.py [hunk 2]
```diff
@@ -43,50 +42,60 @@
...
+            # Only kwargs in the call
+            if "address" in kw:
+                pyclient = self.__address_construction(*a, **kw)
+            else:
+                pyclient = self.__standard_construction(*a, **kw)
         super().__init__(pyclient)
 
-    def __deprecated_construction(self, cluster, address=None, logger_name="Default"):
-        """Initialize a RedisAI client (Deprecated)
+    def __address_construction(self, cluster, address=None, logger_name="Default"):
```
Comment (MattToast) [inline src/python/module/smartredis/client.py:93]: Not desperately important, but if we could retroactively type hint this method that might be nice for future developers
```py
    def __address_construction(
        self,
        cluster: bool,
        address: t.Optional[str] = None,
        logger_name: str = "Default"
    ) -> PyClient:
```

File: src/python/module/smartredis/client.py (hunk 1 above)
Comment (MattToast) [inline src/python/module/smartredis/client.py:124]: Again, not desperately important, but if we could we retroactively type hint this while we are here?
```py
    def __standard_construction(
        self,
        config_options t.Optional[ConfigOptions] = None,
        logger_name: str = "Default"
    ) -> PyClient:
```

**Review by MattToast — APPROVED** [review]
A couple of small change requests but nothing directly related to this PR and not worth holding up approval over. Feel free to handle them here if you'd like or kick them out into new tickets!

LGTM!

File: src/python/module/smartredis/client.py (hunk 1 above)
Comment (billschereriii) [reply src/python/module/smartredis/client.py]: Done

File: src/python/module/smartredis/client.py (hunk 2 above)
Comment (billschereriii) [reply src/python/module/smartredis/client.py]: Done

File: src/python/module/smartredis/client.py (hunk 1 above)
Comment (billschereriii) [reply src/python/module/smartredis/client.py]: Done

**Pull request merged**